odl/solvers/functional/functional.py

```diff
@@ -1303,9 +1318,9 @@ def __init__(self, func):
         if not isinstance(func, Functional):
             raise TypeError('`func` {} is not a `Functional` instance'
                             ''.format(func))
-
+        #HELP: The range should be equal?!
```

Comment:
Actually, I think that here is where it becomes tricky.... if the range is complex then how should one define the convex conjugate? This is related to #590. So my current best answer is: I am not sure of what the range of this functional should be to make most sense.

Reply:
Now we could just check for `functional.range == RealNumbers` and otherwise throw an error

Reply:
Yes so we have that issue already now and just decide not to do anything about it. So IMO let's not raise any exceptions yet, instead do the right thing for `RealNumbers` and otherwise do whatever we were doing all along. At some point this whole complex functionals thing needs a thorough analysis and solution (or maybe not?), but this is not the time.